packages/nango-yaml/lib/parser.ts

```diff
@@ -0,0 +1,75 @@
+import type { NangoModel, NangoYaml, NangoYamlParsed } from '@nangohq/types';
+import { ModelsParser } from './modelsParser.js';
+import { ParserErrorDuplicateModel, ParserErrorMissingId, ParserErrorModelIsLiteral } from './errors.js';
+import type { ParserError } from './errors.js';
+
+export abstract class NangoYamlParser {
+    raw: NangoYaml;
+    parsed: NangoYamlParsed | undefined;
+    modelsParser: ModelsParser;
+    // check that every endpoint is unique across syncs and actions
+    endpoints = new Set<string>();
```

Comment:
why a variable for that while the endpoints can be retrieve looking at the `parsed` content?